Dịch "convexity_vn" - Phần 1 (#1815)
aivivn/d2l-vn: chapter_optimization/convexity_vn.md

This can lead to exciting new optimization variants such as :cite:`Izmailov.Podoprikhin.Garipov.ea.2018`.
-->

Tính lồi đóng một vai trò then chốt trong việc thiết kế các thuật toán tối ưu.
Điều này phần lớn là do các thuật toán sẽ dễ phân tích và kiểm tra hơn trong trường hợp này.
Nói cách khác, nếu thuật toán hoạt động kém ngay cả trong môi trường lồi thì ta không nên kì vọng rằng sẽ thu được kết quả tốt trong môi trường khác.

> Hơn nữa, mặc dù các bài toán tối ưu hóa trong học sâu đa phần là không lồi, chúng lại thường thể hiện một số tính chất lồi gần các cực tiểu cục bộ.

Reviewer: ```suggestion
Hơn nữa, mặc dù các bài toán tối ưu hóa trong học sâu đa phần là không lồi, chúng lại thường thể hiện một số tính chất lồi gần các cực tiểu.
```

Given that, they also need to be contained in $X \cap Y$, thus proving our first theorem.
-->

Chỉ một mình định nghĩa thôi thì sẽ không có tác dụng gì trừ phi bạn làm gì đó với chúng.
Trong trường hợp này, chúng ta có thể nhìn vào hợp và giao của hai tập trong :numref:`fig_convex_intersect`.
Giả sử $X$ và $Y$ là các tập hợp lồi.
Thì $X \cap Y$ cũng sẽ lồi.

> Để thấy được điều này, giả sử rằng bất kì  $a, b \in X \cap Y$. Vì $X$ và $Y$ lồi, đường thẳng chứa $a$ và $b$ sẽ thuộc trong cả $X$ và $Y$.

Reviewer 2: ```suggestion
Để thấy được điều này, hãy xét bất kì $a, b \in X \cap Y$. Vì $X$ và $Y$ lồi, khi đó đoạn thẳng nối $a$ và $b$ sẽ thuộc trong cả $X$ và $Y$.
```

Given that, they also need to be contained in $X \cap Y$, thus proving our first theorem.
-->

Chỉ một mình định nghĩa thôi thì sẽ không có tác dụng gì trừ phi bạn làm gì đó với chúng.
Trong trường hợp này, chúng ta có thể nhìn vào hợp và giao ở :numref:`fig_convex_intersect`.
Giả sử $X$ và $Y$ là các tập hợp lồi.
Thì $X \cap Y$ cũng sẽ lồi.
Để thấy được điều này, giả sử rằng bất kì  $a, b \in X \cap Y$. Vì $X$ và $Y$ lồi, đường thẳng chứa $a$ và $b$ sẽ thuộc trong cả $X$ và $Y$.
Do đó, chúng cũng cần phải thuộc $X \cap Y$, từ đó chứng minh được định lý đầu tiên của chúng ta.

<!--
![The intersection between two convex sets is convex](../img/convex-intersect.svg)
-->

![Giao giữa hai tập hợp lồi cũng là lồi](../img/convex-intersect.svg)
:label:`fig_convex_intersect`

<!--
We can strengthen this result with little effort: given convex sets $X_i$, their intersection $\cap_{i} X_i$ is convex.
To see that the converse is not true, consider two disjoint sets $X \cap Y = \emptyset$.

> Now pick $a \in X$ và $b \in Y$.

Author: cái này là nằm trong comment mà

> Now pick $a \in X$ và $b \in Y$.

Author: à, comment giữ nguyên nhỉ :D để mình commit

The line segment in :numref:`fig_nonconvex` connecting $a$ and $b$ needs to contain some part that is neither in $X$ nor $Y$, since we assumed that $X \cap Y = \emptyset$.
Hence the line segment is not in $X \cup Y$ either, thus proving that in general unions of convex sets need not be convex.
-->

Chúng ta có thể cố thêm một chút nữa để củng cố kết quả này: Cho các tập hợp lồi $X_i$, giao của chúng $\cap_{i} X_i$ là lồi.
Để thấy rằng điều ngược lại là không đúng, hãy xem xét hai tập hợp không giao nhau  $X \cap Y = \emptyset$.
Giờ ta chọn ra $a \in X$ và $b \in Y$.

> Đường thẳng trong :numref:`fig_nonconvex` nối $a$ và $b$ sẽ chứa một vài phần không thuộc cả $X$ và $Y$, vì chúng ta đã giả định rằng $X \cap Y = \emptyset$.

Reviewer 3: ```suggestion
Đoạn thẳng trong :numref:`fig_nonconvex` nối $a$ và $b$ sẽ chứa một vài phần không thuộc cả $X$ và $Y$, vì chúng ta đã giả định rằng $X \cap Y = \emptyset$.
```

> Do đó đoạn thẳng cũng không thuộc $X \cup Y$, từ đó chứng minh rằng: về tổng quát, hợp của các tập hợp lồi không nhất thiết phải lồi.

Reviewer 3: ```suggestion
Do đó đoạn thẳng cũng không  trong $X \cup Y$, từ đó mệnh đề "hợp của các tập lồi nói chung không nhất thiết phải lồi" được chứng minh.
```

<!--
![The union of two convex sets need not be convex](../img/nonconvex.svg)
-->

> ![Hợp của hai tập hợp lồi không nhất thiết phải lồi](../img/nonconvex.svg)

Reviewer 3: ```suggestion
![Hợp của hai tập lồi không nhất thiết phải lồi](../img/nonconvex.svg)
```

Hence the line segment is not in $X \cup Y$ either, thus proving that in general unions of convex sets need not be convex.
-->

Chúng ta củng cố kết quả thêm một chút với mệnh đề: giao của các tập lồi $X_i$ là một tập lồi $\cap_{i} X_i$.

> Để thấy rằng điều ngược lại là không đúng, hãy xem xét hai tập hợp không giao nhau  $X \cap Y = \emptyset$.

Reviewer 2: ```suggestion
Để thấy rằng điều ngược lại là không đúng, hãy xem xét hai tập hợp không giao nhau $X \cap Y = \emptyset$.
```

Giờ ta chọn ra $a \in X$ và $b \in Y$.

> Đoạn thẳng trong :numref:`fig_nonconvex` nối $a$ và $b$ sẽ chứa một vài phần không thuộc cả $X$ và $Y$, vì chúng ta đã giả định rằng $X \cap Y = \emptyset$.

Reviewer 2: ```suggestion
Đoạn thẳng trong :numref:`fig_nonconvex` nối $a$ và $b$ sẽ chứa một vài phần không thuộc cả $X$ hay $Y$, vì chúng ta đã giả định rằng $X \cap Y = \emptyset$.
```

Author: cả thì đi với và là hợp lí rồi mà nhỉ :-/

Reviewer 2: ờ ha, chắc lúc đấy mình đọc nhầm :)).

> Do đó đoạn thẳng cũng không  trong $X \cup Y$, từ đó mệnh đề "hợp của các tập lồi nói chung không nhất thiết phải lồi" được chứng minh.

Reviewer 2: ```suggestion
Do đó đoạn thẳng cũng không nằm trong $X \cup Y$, từ đó chứng minh mệnh đề "hợp của các tập lồi nói chung không nhất thiết phải lồi".
```

In some cases we work with variables of bounded length, such as balls of radius $r$ as defined by $\{\mathbf{x} | \mathbf{x} \in \mathbb{R}^d \text{ and } \|\mathbf{x}\|_2 \leq r\}$.
-->

> Thông thường, các vấn đề trong học sâu đều được định nghĩa trong các miền lồi.

Reviewer 3: ```suggestion
Thông thường, các bài toán trong học sâu đều được định nghĩa trong các miền lồi.
```

> Ví dụ $\mathbb{R}^d$ là tập hợp lồi (xét cho cùng, đoạn thẳng giữa hai điểm bất kỳ thuộc $\mathbb{R}^d$ sẽ thuộc $\mathbb{R}^d$).

Reviewer 3: ```suggestion
Ví dụ $\mathbb{R}^d$ là tập lồi (xét cho cùng, đoạn thẳng giữa hai điểm bất kỳ thuộc $\mathbb{R}^d$ sẽ thuộc $\mathbb{R}^d$).
```

> Trong một vài trường hợp, chúng ta sẽ làm việc với các biến có độ dài cố định, ví dụ như các quả banh có bán kính $r$ được định nghĩa bằng $\{\mathbf{x} | \mathbf{x} \in \mathbb{R}^d \text{ và } \|\mathbf{x}\|_2 \leq r\}$.

Reviewer 3: ```suggestion
Trong một vài trường hợp, chúng ta sẽ làm việc với các biến có biên, ví dụ như các khối cầu có bán kính $r$ được định nghĩa bằng $\{\mathbf{x} | \mathbf{x} \in \mathbb{R}^d \text{ và } \|\mathbf{x}\|_2 \leq r\}$.
```

* Đoàn Võ Duy Thanh
<!-- Phần 1 -->

> * Võ Tấn Phát

Reviewer: ```suggestion
* Võ Tấn Phát
* Lê Khắc Hồng Phúc
* Nguyễn Văn Quang
```
